osu.Game.Rulesets.Taiko/Skinning/LegacyTaikoScroller.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Screens.Play;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Taiko.Skinning
+{
+    public class LegacyTaikoScroller : CompositeDrawable
+    {
+        public LegacyTaikoScroller()
+        {
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load(GameplayBeatmap gameplayBeatmap)
+        {
+            if (gameplayBeatmap != null)
+                ((IBindable<JudgementResult>)LastResult).BindTo(gameplayBeatmap.LastJudgementResult);
+        }
+
+        private bool passing;
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            LastResult.BindValueChanged(result =>
+            {
+                var r = result.NewValue;
+
+                passing = r == null || (r.Judgement.AffectsCombo && r.Type > HitResult.Miss);
```

Comment:
This is also slightly wrong - if the judgement doesn't affect combo it shouldn't affect the value of `passing` in any way (the set should not happen at all).